### PR TITLE
Add Buffer Length Check in ToArray and CopyTo

### PR DESCRIPTION
**Pull request by j0shuams**
In WindowsRuntimeBufferExtension, the existing implementation of ToArray does not allow for creating an empty array, i.e. one of length 0. Yet this behavior is not improper and in certain cases is desirable. We are updating the method to handle this special case. Similarly, the CopyTo method of the same class should support copying to an empty array, from an empty buffer -- for certain special cases. 

This PR...
* updates the ToArray and CopyTo methods to explicitly check if the capacity of the source Buffer is 0
* includes unit tests that check for these edge cases
* moves unit tests out of the net5 specific test file, into the base test file by using the NET5_0 directive
* closes https://github.com/microsoft/CsWinRT/issues/334

### CODE REVIEW

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs [hunk 1]
```diff
@@ -110,6 +110,7 @@ public static void CopyTo(this byte[] source, int sourceIndex, IBuffer destinati
         public static byte[] ToArray(this IBuffer source)
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
+            if (source.Capacity == 0) return Array.Empty<byte>();
```
Comment (manodasanW) [inline cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:113]: capacity or length?

Comment (j0shuams) [reply cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:113]: Buffers can only be constructed with a capacity, so I think the idea is that empty buffers should become empty arrays...

Comment (manodasanW) [reply cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:113]: I would assume a buffer with a capacity of 2 would result in a ToArray of 0 length.

[This](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.windowsruntime.windowsruntimebufferextensions.toarray?view=dotnet-plat-ext-3.1#System_Runtime_InteropServices_WindowsRuntime_WindowsRuntimeBufferExtensions_ToArray_Windows_Storage_Streams_IBuffer_) was the previous documentation for this API which also indicates similarly.

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs [hunk 2]
```diff
@@ -139,13 +141,15 @@ public static void CopyTo(this IBuffer source, byte[] destination)
...
             if (destination == null) throw new ArgumentNullException(nameof(destination));
+            if (source.Length == 0 && destination.Length == 0) return;
 
             CopyTo(source, 0, destination, 0, checked((int)source.Length));
         }
 
 
         public static void CopyTo(this IBuffer source, uint sourceIndex, byte[] destination, int destinationIndex, int count)
         {
+            if (source.Length == 0 && destination.Length == 0) return;
```
Comment (manodasanW) [inline cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:152]: You should move this after the null checks, otherwise we can get null pointer here.

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs [hunk 3]
```diff
@@ -118,6 +119,7 @@ public static byte[] ToArray(this IBuffer source)
         public static byte[] ToArray(this IBuffer source, uint sourceIndex, int count)
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
+            if (source.Length == 0) return Array.Empty<byte>();
```
Comment (manodasanW) [inline cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:122]: I feel like this scenario should technically error.  We are told to convert to an array with the array starting with data from source index to a length of count.   In the length == 0 case and count != 0, we return a empty array which might unexpected to the caller and maybe even against what the API was documented as.

We should probably do source.length == 0 and count == 0, return empty.  otherwise throw exception like before.

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs [hunk 4]
```diff
@@ -139,13 +141,15 @@ public static void CopyTo(this IBuffer source, byte[] destination)
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (destination == null) throw new ArgumentNullException(nameof(destination));
+            if (source.Length == 0 && destination.Length == 0) return;
```
Comment (manodasanW) [inline cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:144]: I think we probably only need to check source.length==0 here.

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs (hunk 2 above)
Comment (manodasanW) [reply cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:152]: We also probably should do source.length == 0 && count == 0 rather than destination.length

**Review by Scottj1s — APPROVED** [review]

**Comment by Scottj1s** [comment]
while you're in this code, maybe also take care of #497

File: cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs [hunk 5]
```diff
@@ -140,6 +141,9 @@ public static void CopyTo(this IBuffer source, byte[] destination)
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (destination == null) throw new ArgumentNullException(nameof(destination));
 
+            // If buffer and array are empty, nothing to copy
```
Comment (manodasanW) [inline cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs:144]: buffer is empty

**Review by manodasanW — APPROVED** [review]
